streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestDriver.java

```diff
@@ -357,256 +274,197 @@ public static void verify(final String kafka, final Map<String, Set<Integer>> al
-                final int expected = max - min;
-                if (entry.getValue() == null || expected != entry.getValue()) {
-                    if (print) {
-                        System.out.println("fail: key=" + entry.getKey() + " dif=" + entry.getValue() + " expected=" + expected);
-                    }
-                    return false;
-                }
-            }
+        VerificationResult(final boolean passed, final String result) {
+            this.passed = passed;
+            this.result = result;
         }
-        return true;
-    }
 
-    private static boolean verifyCnt(final Map<String, Long> map, final Map<String, Set<Integer>> allData, final boolean print) {
-        if (map.isEmpty()) {
-            if (print) {
-                System.out.println("cnt is empty");
-            }
-            return false;
-        } else {
-            if (print) {
-                System.out.println("verifying cnt");
-            }
+        public boolean passed() {
+            return passed;
+        }
 
-            if (map.size() != allData.size()) {
-                if (print) {
-                    System.out.println("fail: resultCount=" + map.size() + " expectedCount=" + allData.size());
-                }
-                return false;
-            }
-            for (final Map.Entry<String, Long> entry : map.entrySet()) {
-                final int min = getMin(entry.getKey());
-                final int max = getMax(entry.getKey());
-                final long expected = (max - min) + 1L;
-                if (expected != entry.getValue()) {
-                    if (print) {
-                        System.out.println("fail: key=" + entry.getKey() + " cnt=" + entry.getValue() + " expected=" + expected);
-                    }
-                    return false;
-                }
-            }
+        public String result() {
+            return result;
         }
-        return true;
     }
 
-    private static boolean verifySum(final Map<String, Long> map, final Map<String, Set<Integer>> allData, final boolean print) {
-        if (map.isEmpty()) {
-            if (print) {
-                System.out.println("sum is empty");
-            }
+    private static VerificationResult verifyAll(final Map<String, Set<Integer>> inputs,
+                                                final Map<String, Map<String, LinkedList<Number>>> events) {
+        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        final PrintStream resultStream = new PrintStream(byteArrayOutputStream);
+        boolean pass;
+        pass = verifyTAgg(resultStream, inputs, events.get("tagg"));
+        pass &= verify(resultStream, "min", inputs, events, SmokeTestDriver::getMin);
+        pass &= verify(resultStream, "max", inputs, events, SmokeTestDriver::getMax);
+        pass &= verify(resultStream, "dif", inputs, events, key -> getMax(key).intValue() - getMin(key).intValue());
```

Comment:
Is it a good idea to convert `long` to `int` here? Are we sure it's zero risk? Also, it it future proof if we change the test data with potential bigger values that don't fit into ints?

Reply:
It's not converting long to int. It's reading a `Number` as an int. Since we populated the `Number` with an int to begin with, it should be perfectly safe.